CMake/ITKModuleMacros.cmake

```
@@ -471,16 +471,7 @@ macro(itk_module_add_library _name)
     set(_LIBRARY_BUILD_TYPE)
   endif()
   add_library(${_name} ${_LIBRARY_BUILD_TYPE} ${ARGN})
-  target_compile_features(${_name} PUBLIC cxx_nullptr
-                                          cxx_override
-                                          cxx_constexpr
-                                          cxx_range_for
-                                          cxx_rvalue_references
-                                          cxx_static_assert
-                                          cxx_strong_enums
-                                          cxx_lambdas
-                                          cxx_noexcept
-                                          cxx_alias_templates )
+  target_compile_features(${_name} PUBLIC cxx_std_${CMAKE_CXX_STANDARD})
```

Does ITK compiled in C++17 force consumers to also use C++17?

Maybe

```suggestion
  target_compile_features(${_name}
    PRIVATE cxx_std_${CMAKE_CXX_STANDARD}
    INTERFACE cxx_std_14)
```

---

I had the second option you propose as my first attempt at this patch.  Discussion indicated that this PR should be modified to follow the standard that is chosen by the user.

This may be overly strict but would have the least surprises.